webapp/src/extensionsBrowser.tsx

```diff
@@ -17,6 +17,7 @@ const emptyCard: EmptyCard = { name: "", loading: true }
 
 interface ExtensionsProps {
     hideExtensions: () => void;
+    parent: pxt.editor.IProjectView;
```

Comment:
Don't pass in the IProjectView. It makes the coupling too tight so it's hard to refactor this in the future. Instead, pass in a callback that calls the specific method you need.

Reply:
This is now a callback as suggested.